Sources/PostgresNIO/New/PreparedStatement.swift

```diff
@@ -0,0 +1,40 @@
+/// A prepared statement.
+///
+/// Structs conforming to this protocol will need to provide the SQL statement to
+/// send to the server and a way of creating bindings are decoding the result.
+///
+/// As an example, consider this struct:
+/// ```swift
+/// struct Example: PostgresPreparedStatement {
+///     static var sql = "SELECT pid, datname FROM pg_stat_activity WHERE state = $1"
+///     typealias Row = (Int, String)
+///
+///     var state: String
+///
+///     func makeBindings() -> PostgresBindings {
+///         var bindings = PostgresBindings()
+///         bindings.append(.init(string: self.state))
```

Comment:
```suggestion
///         bindings.append(self.state)
```
does this work as well?

Reply:
No, I get back an error: `Cannot convert value of type 'String' to expected argument type 'PostgresData'`

Reply:
Oh, we shouldn't create `PostgresData`. The problem here is that we better names for those methods, so that users can use them without Interpolation as well.

```swift
struct PostgresBindings {
    @inlinable
    public mutating func appendInterpolation<Value: PostgresEncodable>(_ value: Value) throws {
        try self.binds.append(value, context: .default)
        self.sql.append(contentsOf: "$\(self.binds.count)")
    }

    @inlinable
    public mutating func appendInterpolation<Value: PostgresEncodable>(_ value: Optional<Value>) throws {
        switch value {
        case .none:
            self.binds.appendNull()
        case .some(let value):
            try self.binds.append(value, context: .default)
        }

        self.sql.append(contentsOf: "$\(self.binds.count)")
    }
}
```